plugins/source/gitlab/codegen/recipes/base.go

```diff
@@ -0,0 +1,217 @@
+package recipes
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"go/format"
+	"os"
+	"path"
+	"reflect"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/cloudquery/plugin-sdk/codegen"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/iancoleman/strcase"
+)
+
+//go:embed templates/*.go.tpl
+var templatesFS embed.FS
+
+type Resource struct {
+	Service        string
+	SubService     string
+	ServicePath    string
+	GlobalResource bool
+	ServiceFunc    interface{}
```

Comment:
use `any` instead of `interface{}` (it's an alias, just increases code readability)